### PR TITLE
fix: find one profile

### PR DESCRIPTION
**Pull request by olugbenga1**
# What does this PR do?
This PR fixes the error message and HTTP status code returned when an invalid user id is parsed when fetching user profile. In the current implementation, an invalid user id when fetching the user's profile returns a 500 HTTP status code and an internal server error message:
* Error Handling: This fix returns specific HTTP status codes and messages based on various scenarios.

# Checklist:
- [x] Implemented GET /api/v1/profile/:id to retrieve the current user's profile settings.
- [x] Return 400 bad request if user id is invalid
- [x] Tested endpoint functionality to ensure it correctly handles various scenarios.
- [x] Added appropriate HTTP status codes and response messages for edge cases.
- [x] Added unit tests for the `Profile Service`.

# Response Examples
**Invalid user id**

```json
{
    "message": "Invalid user ID",
    "error": "Bad Request",
    "statusCode": 400
}
```

# Link to Issue
Link to this issue: [FEAT Create API Endpoints to Retrieve User Profile Settings](https://github.com/hngprojects/hng_boilerplate_nestjs/issues/122)

### CODE REVIEW

File: src/modules/profile/profile.service.ts
Comment (Heba-WebDev) [inline src/modules/profile/profile.service.ts:1]: Please use the CustomHttpException, where you're throwing an exception.

**Comment by Heba-WebDev** [comment]
Please update the PR title, all lower case

File: src/modules/profile/profile.service.ts [hunk 1]
```diff
@@ -14,34 +15,31 @@ export class ProfileService {
...
+    }
 
-      const profile = userProfile.profile;
-      if (!profile) {
-        throw new NotFoundException('Profile not found');
-      }
+    const user = await this.userRepository.findOne({ where: { id: userId } });
+    if (!user) {
+      throw new NotFoundException('User not found');
+    }
```
Comment (davidolanrewaju) [inline src/modules/profile/profile.service.ts:25]: There's a global exception called CustomHttpException for handling errors, please make use of it make use of

File: src/modules/profile/profile.service.ts [hunk 2]
```diff
@@ -14,34 +15,31 @@ export class ProfileService {
...
 
-      return responseData;
-    } catch (error) {
-      if (error instanceof NotFoundException || error instanceof BadRequestException) {
-        throw error;
-      }
-      throw new InternalServerErrorException(`Internal server error: ${error.message}`);
+    const profile = userProfile.profile;
+    if (!profile) {
+      throw new NotFoundException('Profile not found');
```
Comment (davidolanrewaju) [inline src/modules/profile/profile.service.ts:34]: Apply the global custom error exception

File: src/modules/profile/profile.service.ts [hunk 3]
```diff
@@ -14,34 +15,31 @@ export class ProfileService {
...
-      if (!user) {
-        throw new NotFoundException('User not found');
-      }
-
-      const userProfile = await this.userRepository.findOne({
-        where: { id: userId },
-        relations: ['profile'],
-      });
+    if (!isUUID(userId, '4')) {
+      throw new BadRequestException('Invalid user ID');
```
Comment (davidolanrewaju) [inline src/modules/profile/profile.service.ts:19]: Apply the global custom exception error handler

File: src/modules/profile/profile.service.ts (hunk 2 above)
Comment (olugbenga1) [reply src/modules/profile/profile.service.ts]: Done

File: src/modules/profile/profile.service.ts (hunk 3 above)
Comment (olugbenga1) [reply src/modules/profile/profile.service.ts]: Done

File: src/modules/profile/profile.service.ts (hunk 1 above)
Comment (olugbenga1) [reply src/modules/profile/profile.service.ts:26]: Done

**Review by Heba-WebDev — APPROVED** [review]

**Review by TheCodeGhinux — APPROVED** [review]

**Review by TheCodeGhinux — DISMISSED** [review]

**Review by davidolanrewaju — APPROVED** [review]

File: src/modules/profile/profile.service.ts [hunk 4]
```diff
@@ -14,63 +16,76 @@ export class ProfileService {
...
+    if (!user || !user.profile) {
+      throw new CustomHttpException('User profile not found', 404);
     }
+
+    const userProfile = await this.profileRepository.findOne({
+      where: { id: user.profile.id },
+    });
+
+    if (!userProfile) {
+      throw new CustomHttpException('Profile not found', 404);
```
Comment (Cobidex) [inline src/modules/profile/profile.service.ts:80]: Use constants

**Review by Cobidex — APPROVED** [review]

**Review by TheCodeGhinux — DISMISSED** [review]

**Review by TheCodeGhinux — APPROVED** [review]

**Review by TheCodeGhinux — APPROVED** [review]

File: src/modules/profile/profile.controller.ts [hunk 5]
```diff
@@ -1,11 +1,13 @@
...
+import { Controller, Get, Post, Body, Patch, Param, Delete, UseInterceptors } from '@nestjs/common';
 import { ProfileService } from './profile.service';
 import { ApiBearerAuth, ApiOperation, ApiResponse, ApiTags } from '@nestjs/swagger';
 import { UpdateProfileDto } from './dto/update-profile.dto';
+import { ResponseInterceptor } from 'src/shared/inteceptors/response.interceptor';
 
 @ApiBearerAuth()
 @ApiTags('Profile')
 @Controller('profile')
+@UseInterceptors(ResponseInterceptor)
```
Comment (davidolanrewaju) [inline src/modules/profile/profile.controller.ts:10]: Please remove this userInterceptor if it's not doing anything

**Review by TheCodeGhinux — APPROVED** [review]

**Review by TheCodeGhinux — APPROVED** [review]

**Review by TheCodeGhinux — DISMISSED** [review]

**Review by TheCodeGhinux — DISMISSED** [review]
Good
